### ISSUE TITLE
TypeError: data type "c1" not understood

### ISSUE THREAD

**Issue by assadRasheed** [tartavull/snn-rl#11]
i am unable to sovle this error .why this error  is show ing 
TypeErrorTraceback (most recent call last)
<ipython-input-1-97767a7530eb> in <module>()
     20 timeAndRefrac = timeAndRefrac
     21 
---> 22 class gupta_paper:
     23 	'''
     24                 Main program variables are set the simulation is run.  Specific neuron models are

<ipython-input-1-97767a7530eb> in gupta_paper()
     29         neuralnet = Network()
     30         dictionary = dictionary()
---> 31         spiketimes = dictionary.spikeTimes(dictionaryLongitude, spikeInterval, testingSpikesPerChar, testingEpochs)
     32         trainingSpikeTimes = dictionary.spikeTimes(dictionaryLongitude, spikeInterval, trainingSpikesPerChar, trainingEpochs)
     33         LIK = SpikeGeneratorGroup(N=15, indices=spiketimes[:,0], times=spiketimes[:,1]*ms)

C:\Users\zaiaad-pc\AnacondaProjects_old\snn-rl-master\furtherFormulas\dictionary.py in spikeTimes(self, dictionaryLongitude, spikeInterval, spikesPerChar, epochs)
     13         def spikeTimes(self, dictionaryLongitude, spikeInterval, spikesPerChar, epochs):
     14 
---> 15                 dictionary = zeros(26, dtype='c1, (5,3)u1')
     16 		dictionary[:] =[('A',[[1, 0, 1],[0, 1, 0],[0, 0, 0],[0, 1, 0],[0, 1, 0]]),
     17                                             ('B',[[0, 0, 1],[0, 1, 0],[0, 0, 1],[0, 1, 0],[0, 0, 1]]),

TypeError: data type "c1" not understood

**Comment by nmsutton** [comment]
Apologies for taking a while to respond to this. I was able to recreate this error. I did some digging and found it was an issue due to a later numpy version (similar to [this](https://github.com/statsmodels/statsmodels/issues/3841)). I recoded the dictionary array in the dictionary file and stopped the error. Thank you for reporting it.

While testing the code I also found a matplotlib newer version error occuring too. I found a solution [here](https://groups.google.com/a/continuum.io/forum/#!topic/anaconda/gwQkR5gosTw). I have uploaded a new dictionary file with the fixes to the the project. I have been able to run the software to completion after adding the fixes. I tested this with a clean install of anaconda 2.5.0 (haven’t tested the latest anaconda) on Ubuntu 16.04.1 LTS and installing brian2 (2.0.2.1) through instuctions similar to [these](https://anaconda.org/brian-team/brian2). When you use the new dictionary.py file are you able to run the software successfully?
